File: typescript/docs_code_snippets/traces-get.ts
import { Opper } from "opperai";

const opper = new Opper();

const trace = await opper.traces.getTrace("trace-id");
console.log(`Name: ${trace.name}, Spans: ${trace.span_count}`);
for (const s of trace.spans) {
  const indent = s.parent_id ? "    " : "  ";
  console.log(`${indent}${s.name} (${s.id.slice(0, 8)}...)`);
}

File: python/docs_code_snippets/traces-get.py
import time
from datetime import datetime, timezone
from opperai import Opper

opper = Opper()

# Setup: create a span (which creates a trace)
span = opper.spans.create(
    name="my-pipeline",
    start_time=datetime.now(timezone.utc).isoformat(),
)
time.sleep(1)  # wait for trace to be indexed
# /setup

# --- docs ---
trace = opper.traces.get(span.trace_id)
print(f"Name: {trace.name}, Spans: {trace.span_count}")
for s in trace.spans:
    indent = "  " if not s.parent_id else "    "
    print(f"{indent}{s.name} ({s.id[:8]}...)")
# --- /docs ---

# Cleanup
opper.traces.delete(span.trace_id)
